### PR TITLE
Add color support

### PR DESCRIPTION
**Pull request by Hdt80bro**
`EnumColor` may want to instead be moved to a different file, rather than being shoved to the bottom.

### CODE REVIEW

**Comment by Garanas** [comment]
Great find - wasn't aware it was possible beyond the primary colors.

**Comment by Garanas** [comment]
Great find - wasn't aware it was possible beyond the primary colors. I've added EnumColor as a type, what are your thoughts on that?

**Comment by Hdt80bro** [comment]
I did that at first, but realized that it clogs up the type information, and for some reason adds the comment on `Color` to everything.
image
I think this is fine
image
Since hexcodes are strings, I figure the type information of `EnumColor` is already a subset of that, so we don't loose documentation.
However, if we could somehow specify a formatted string as a type, I'd be open to that. We could do something like
```
---@alias Color OpaqueColor | TransparentColor | EnumColor
---@alias OpaqueColor string["XXXXXX"] # somehow specify a string of length 6 containing only hex characters 
---@alias TransparentColor  string["XXXXXXXX"] # same, but for length 8
---@alias EnumColor # all 140 options...
```
Nevertheless, we still couldn't keep it from expanding since it's an `@alias` not a `@class`, and we couldn't make it a `@class` because those can't extend literals or aliases.

TL;DR I think it'd be better to leave it in the description comment because it's cleaner
